source/UberCore/Networking/APIEndpoint.swift

```diff
@@ -22,32 +22,51 @@
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
 
+/**
+ *  Protocol for all endpoints to conform to.
+ */
+public protocol APIEndpoint {
+    var body: Data? { get }
+    var headers: [String: String]? { get }
+    var host: String { get}
+    var method: HTTPMethod { get }
+    var path: String { get }
+    var query: [URLQueryItem] { get }
+}
 
-import XCTest
-@testable import UberRides
-
-class UberRidesTests: XCTestCase {
-    
-    override func setUp() {
-        super.setUp()
-        // Put setup code here. This method is called before the invocation of each test method in the class.
-    }
-    
-    override func tearDown() {
-        // Put teardown code here. This method is called after the invocation of each test method in the class.
-        super.tearDown()
+public extension APIEndpoint {
+    var body: Data? {
+        return nil
     }
-    
-    func testExample() {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct results.
+
+    var headers: [String: String]? {
+        return nil
     }
-    
-    func testPerformanceExample() {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
+
+    var host: String {
+        if Configuration.shared.isSandbox {
+            return "https://sandbox-api.uber.com"
```

Comment:
That's a good point. This move was just a rename, so this is kept from the old `UberAPI.swift` file. 

This would be better as a `Configuration.baseURL` property. I've logged it under #194.